### ISSUE TITLE
Can we use Riverpod's suggested NotifierProvider + code-generation with this plugin?

### ISSUE THREAD

**Issue by arnab** [quantosapplications/riverpod_infinite_scroll#36]
I would like to use the Riverpod (suggested) generated approach to build the `NotifierProvider`: 
* https://docs-v2.riverpod.dev/docs/providers/state_notifier_provider (see the callout: Prefer using [NotifierProvider](https://docs-v2.riverpod.dev/docs/providers/notifier_provider) instead.
* https://docs-v2.riverpod.dev/docs/providers/notifier_provider
* https://docs-v2.riverpod.dev/docs/about_code_generation

Could you please show some example code to make this possible? Or is this a feature request?

**Comment by SasLuca** [comment]
I am curious about this too, hoping to get some info on this, would be nice to know.
If need be we could maybe work on a pull request for this if @ftognetto is too busy.

**Comment by ftognetto** [comment]
Hi everyone, yes, it's actually a useful feature to add but I haven't had time to work on it yet. Surely a pull request would also be excellent as a starting point and I could spend some time on it in August. Thank you all

**Comment by flikkr** [comment]
@ftognetto has this been implemented?

**Comment by ftognetto** [comment]
Not really, I myself am using riverpod generation with riverpod_infinite_scroll but the new version of riverpod contains many sealed classes and I'm stuck at creating a clean api.
It would be great to have an help on this, I can publish what I've done so far on a separate branch
